core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -86,18 +394,175 @@ public TableMetadata getTableMetadata(String namespace, String table) throws Exe
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType, (int) ru));
+        putScalingPolicyRequestList.add(
+            buildPutScalingPolicyRequest(
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType));
+      }
+    }
+
+    // request
+    for (RegisterScalableTargetRequest registerScalableTargetRequest :
+        registerScalableTargetRequestList) {
+      try {
+        applicationAutoScalingClient.registerScalableTarget(registerScalableTargetRequest);
+      } catch (Exception e) {
+        throw new ExecutionException(
+            "Unable to register scalable target for " + registerScalableTargetRequest.resourceId(),
+            e);
+      }
+    }
+
+    for (PutScalingPolicyRequest putScalingPolicyRequest : putScalingPolicyRequestList) {
+      try {
+        applicationAutoScalingClient.putScalingPolicy(putScalingPolicyRequest);
+      } catch (Exception e) {
+        throw new ExecutionException(
+            "Unable to put scaling policy request for " + putScalingPolicyRequest.resourceId(), e);
+      }
+    }
+  }
+
+  private RegisterScalableTargetRequest buildRegisterScalableTargetRequest(
+      String resourceID, String type, int ruValue) {
+    return RegisterScalableTargetRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAPPING.get(type))
+        .minCapacity(ruValue > 10 ? ruValue / 10 : ruValue)
+        .maxCapacity(ruValue)
+        .build();
+  }
+
+  private PutScalingPolicyRequest buildPutScalingPolicyRequest(String resourceID, String type) {
+    return PutScalingPolicyRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAPPING.get(type))
+        .policyName(getPolicyName(resourceID, type))
+        .policyType(PolicyType.TARGET_TRACKING_SCALING)
+        .targetTrackingScalingPolicyConfiguration(getScalingPolicyConfiguration(type))
+        .build();
+  }
+
+  private String getPolicyName(String resourceID, String type) {
+    return resourceID + "-" + type;
+  }
+
+  private String getTableResourceID(String namespace, String table) {
+    return "table/" + getFullTableName(namespacePrefix, namespace, table);
+  }
+
+  private String getGlobalIndexResourceID(String namespace, String table, String globalIndex) {
+    return "table/"
+        + getFullTableName(namespacePrefix, namespace, table)
+        + "/index/"
+        + getGlobalIndexName(namespace, table, globalIndex);
+  }
+
+  private TargetTrackingScalingPolicyConfiguration getScalingPolicyConfiguration(String type) {
+    return TargetTrackingScalingPolicyConfiguration.builder()
+        .predefinedMetricSpecification(
+            PredefinedMetricSpecification.builder()
+                .predefinedMetricType(
+                    type.contains("read")
```

Comment:
This `type` always be "read" or "write". No "index-read" is coming here?

Reply:
It uses `index-read` here. Thanks for your suggestions. I change to use `Map`, applied in [166ec54](https://github.com/scalar-labs/scalardb/pull/301/commits/166ec546eb44ccd9bdc33932e43e2deabfb79cda), other suggestions also were applied as well